riptide-opentracing/src/main/java/org/zalando/riptide/opentracing/CompositeLifecyclePolicy.java

```diff
@@ -4,29 +4,27 @@
 import io.opentracing.Tracer;
 import lombok.AllArgsConstructor;
 import org.zalando.riptide.RequestArguments;
-import org.zalando.riptide.opentracing.span.SpanDecorator;
 
-import javax.annotation.Nullable;
 import java.util.Collection;
+import java.util.Optional;
 import java.util.function.Consumer;
 
 @AllArgsConstructor
 final class CompositeLifecyclePolicy implements LifecyclePolicy {
 
     private final Collection<LifecyclePolicy> policies;
 
-    @Nullable
     @Override
-    public Span start(final Tracer tracer, final RequestArguments arguments, final SpanDecorator decorator) {
+    public Optional<Span> start(final Tracer tracer, final RequestArguments arguments) {
         for (final LifecyclePolicy policy : policies) {
-            @Nullable final Span span = policy.start(tracer, arguments, decorator);
+            final Optional<Span> span = policy.start(tracer, arguments);
 
-            if (span != null) {
-                return new FinishingSpan(span, policy::finish);
+            if (span.isPresent()) {
```

Comment:
seems to me that span.map(s -> new FinishingSpan(s, policy::finish)) would be shorter here

Reply:
I'd still need to use `if (x.isPresent()) {return x.get();}` afterwards since I need to return from the outer method which I can't do within a lambda.

Reply:
ah, missed that return will be executed only in case of presence